src/lib.rs

```
@@ -497,6 +559,40 @@ fn g_expr(token_list: &[Token]) -> Result<IntermediateResult, CalcError> {
     if !token_list.is_empty() {
         match token_list[0] {
             Token::Number(n) => Ok(IntermediateResult::new(n, 1)),
+            Token::Cosine | Token::Sine | Token::Tangent | Token::Logarithm => {
+                if token_list[1] == Token::OpenParen {
+                    let ir = d_expr(&token_list[2..])?;
+                    let close_paren = ir.tokens_read + 2;
+                    if close_paren < token_list.len() {
+                        match token_list[close_paren] {
+                            Token::CloseParen => {
+                                let mut val = 0.0; //TODO: not declare this val here like the warningwants
+                                match token_list[0] {
+                                    Token::Cosine => val = ir.value.cos(),
+                                    Token::Sine => val = ir.value.sin(),
+                                    Token::Tangent => val = ir.value.tan(),
+                                    Token::Logarithm => {
+                                        val = ir.value.log(10.0)
+                                    }
+                                    _ => val = 0.0, //TODO: maybe get this outta the way too
```

This ought to be an `unreachable!()` versus silently returning zero.